src/Microsoft.DotNet.Interactive.PowerShell/PowerShellExtensions.cs

```diff
@@ -27,6 +28,21 @@ public static void InvokeAndClear(this PowerShell pwsh)
         }
     }
 
+    public static Collection<PSObject> InvokeAndClearWithResult(this PowerShell pwsh)
```

Comment:
Since this method is identical behaviorally to `InvokeAndClear`, I would suggest just changing the signature of `InvokeAndClear` to return the result. This will reduce some duplication and the risk that these two methods might eventually diverge in behavior.

Reply:
running tests